src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py

```diff
@@ -16,12 +16,11 @@ def get_device_id_completion_list(prefix, action, parsed_args, **kwargs):#pylint
     client = iot_hub_service_factory(kwargs)
     return [d.device_id for d in iot_device_list(client, parsed_args.hub_name, top=100)] if parsed_args.hub_name else []
 
-register_cli_argument('iot hub', 'hub_name', options_list=('--name', '-n'),
-                      completer=get_resource_name_completion_list('Microsoft.Devices/IotHubs'),
-                      help='The IoT Hub name.')
-register_cli_argument('iot device', 'hub_name', options_list=('--hub-name', '--hub'),
-                      completer=get_resource_name_completion_list('Microsoft.Devices/IotHubs'),
-                      help='The IoT Hub name.')
+hub_name_type = CliArgumentType(completer=get_resource_name_completion_list('Microsoft.Devices/IotHubs'),
+                                help='IoT Hub name.')
+
+register_cli_argument('iot hub', 'hub_name', hub_name_type, options_list=('--name', '-n'))
+register_cli_argument('iot device', 'hub_name', hub_name_type, options_list=('--hub-name', '--hub'))
```

Comment:
Already updated in my latest commit. Please check it out. Thanks!

---
In reply to: [85955221](https://github.com/Azure/azure-cli/pull/1188#discussion_r85955221) [](ancestors = 85955221)